include/SimCore/PluginFactory.h

```diff
@@ -4,33 +4,36 @@
 #include <variant>
```

Comment:
This is getting bloated. I think I will replace this with a templated factory as done [in fire](https://github.com/LDMX-Software/fire/blob/trunk/include/fire/factory/Factory.h). I may need to add a vector holding the created objects so many different locations can access the list of manufactured objects.